resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -71,8 +68,9 @@
         id = parent.queryContextManager.add(this);
 
         if (parent.isOptResourceEnabled()) {
-            optResource = new DefaultDnsRawRecord(
-                    StringUtil.EMPTY_STRING, DnsRecordType.OPT, parent.maxPayloadSize(), 0, Unpooled.EMPTY_BUFFER);
+            optResource = new AbstractDnsOptPseudoRrRecord(parent.maxPayloadSize(), 0, 0) {
+                // This is somewhat of a hack and should be removed later on.
```

Comment:
can you describe this in more detail ... what is a hack? maybe create a followup issue?